Demo tinygo CI: add tinygo unit test runner [WON'T MERGE]

leongross opened the pull request:
Work in progress test to see to analyze tinygo behavior when running the tests for `pkg` and `cmds`.

leongross committed:
chore: typo

Signed-off-by: leongross <leon.gross@9elements.com>

leongross committed:
ci: add tinygo test runner

Signed-off-by: leongross <leon.gross@9elements.com>

rminnich reviewed (commented):
can you expand on what a cmdlet is?

leongross committed:
ci: update tinygo version

PR https://github.com/tinygo-org/tinygo/pull/4736 adds runtime.Goexit to {Skip,Fail}Now,
which currently fails in the go tests.

Signed-off-by: leongross <leon.gross@9elements.com>

jensdrenhaus commented:
This is to demonstrate the errors when running the u-root unit test with tinygo compiler.
It does not make sense to merge yet, as there is quite some work to do until all u-root commands are runnable when compiled with tinygo compiler.

MDr164 commented:
No response from @jenstopp so I assume we can close it for now. Feel free to re-open as needed.
